ADR/ADR028-use-notify-for-users.md

```diff
@@ -0,0 +1,24 @@
+# ADR029: Use GOV.UK Notify for emails sent to users for simple notifications
+
+Date: 2024-03-28
+
+## Status
+
+Accepted
+
+
+## Context
+
+[Submission emails](https://github.com/alphagov/forms-runner/blob/main/app/mailers/form_submission_mailer.rb) and [submission confirmation emails](https://github.com/alphagov/forms-runner/blob/main/app/mailers/form_submission_confirmation_mailer.rb) in runner and the [OTP to verify a change in submission email via admin](https://github.com/alphagov/forms-admin/blob/main/app/mailers/submission_email_mailer.rb) are sent via GOV.UK Notify. Notifications to users relating to account management (such as when they are upgraded from a trial to full user account), will be sent using GOV.UK Notify.
```

Comment:
I think we can remove the links to our source code since we're considering ADR immutable and those links could break with a bit of refactoring etc. Few minor suggestions, happy to discuss.

<blockquote>

This is an addendum to [ADR019](ADR019-user-amazon-ses.md) which set out the intention to use Amazon's Simple Email Service instead of GOV.UK Notify to send emails.

GOV.UK Forms has continued to use GOV.UK Notify to send emails including:
- submission confirmation emails to members of the public who have completed a form.
- the submitted forms to the service processing them. 
- One Time Passcodes to verify the ownership of the submission email address when creating or editing a form.

Amazon SES is currently used during the authentication process when logging into the form builder via Auth0, a third party authentication service, to send One Time Passcodes since Auth0 does not offer integration with the GOV.UK Notify API. Using Amazon SES also provides enforced TLS.

GOV.UK Forms now needs to send notifications to users relating to account management (such as when they are upgraded from a trial to full user account).
</blockquote>